source/MaterialXCore/Document.cpp

```diff
@@ -173,6 +173,10 @@ void Document::importLibrary(const ConstDocumentPtr& library, const CopyOptions*
         {
             childCopy->setColorSpace(library->getColorSpace());
         }
+        if (!childCopy->hasUnit() && library->hasUnit())
+        {
+            childCopy->setUnit(library->getUnit());
```

Comment:
This will work for a single length unit set, but that about having multiple unit types defined in the document? For example, both _length_ and _angle_. Maybe we have not settled for a syntax on how to define that yet?

Reply:
There are two cases. 
1. This is for non-Document Elements:
Here it does't make sense to have an Element have > 1 unit specifier. e.g. You can't specify unit to be "meter" + "kelvin".
2. For the root / Document Element, this "unit" attribute has the suggestion to keep multiple comma separated units, but await feedback on this. Unlike CM which has a single target "colorspace" attribute, the proposal was to have multiple units for the "unit" attribute at the Document level. This conflicts with the requirement that there be only one unit for non Document Elements. Thus the suggestion to have a different name for the root or have multiple attributes -- or we leave it as it is but treat Document Elements as a special case.

A note here that work to fix up what happens to Document attributes on inputs is work in progress as the information is essentially "lost". i.e. This code doesn't even deal with Document level attributes ("version, namespace, geomprefix, colorspace, and for here unit).

Reply:
As mentioned in discussion we will not have multiple unit types as targets on the document as a unit target at the document level doesn't really solve the issue of per geometry units which is probably best set at the `geometry` or `collection` level, or globally.

Reply:
Then should we set this at all for units here? For any imported child that doesn't have a unit set it sets the unit from the library?? This loops over root level children, so as far as I understand it's setting the units on any material, look, nodedef, nodegraph, node, etc. that got imported. Should we not only have units on value elements? (inputs and parameters?)

Reply:
This should be removed as we're not supporting units on documents.